app/javascript/components/decks/BuildTab.tsx

```diff
@@ -1,17 +1,54 @@
-import { Text, Input, ButtonGroup, Button, Divider } from "@chakra-ui/react"
+import { Text, Input, Image, ButtonGroup, Button, Divider, Box, chakra, Link, ListItem, Table, TableCaption, Tbody, Td, Tfoot, Th, Thead, Tr, useColorModeValue, useDisclosure, Modal, HStack, ModalBody, ModalCloseButton, ModalContent, ModalFooter, ModalHeader, ModalOverlay, Tooltip, Stack, Spacer } from "@chakra-ui/react"
 import * as React from "react"
 
-import { useFilter } from '../../hooks/useFilter'
+import { stringInclude, useFilter } from '../../hooks/useFilter'
+import { Card } from "../Card"
+import { FactionIcon } from '../FactionIcon'
+import { WithModal } from "./WithModal"
+import { WithPopover } from "./WithPopover"
+import { sortBy } from 'lodash'
+import { TypeIcon } from "../TypeIcon"
 
-export const BuildTab = ({ cards, handleAdd, handleRemove }) => {
+export const BuildTab = ({ deckbox, cards, handleSetSlot }) => {
+  // typeahead results specifically
   const [results, setResults] = React.useState<Card[]>([])
   const [query, setQuery] = React.useState("")
   const [cursor, setCursor] = React.useState(0)
 
+  // cards listed in the main tray
+  const [selectedFactions, setSelectedFactions] = React.useState({
```

Comment:
could make a `useToggleMap` hook to dry this up

Reply:
maybe it just takes a string[], and we make the underlying map and useState under the covers, and just expose the [selectedX, toggleX]